src/DPGO_utils.cpp

```diff
@@ -238,20 +236,48 @@ std::vector<RelativeSEMeasurement> read_g2o_file(const std::string &filename,
     } else if ((token == "VERTEX_SE2") || (token == "VERTEX_SE3:QUAT")) {
       // This is just initialization information, so do nothing
       continue;
+    } else if ((token == "FIX")) {
+      LOG(WARNING) << "[read_g2o_file] FIX ID_SET is not supported. Skipping line...";
+      continue;
     } else {
-      LOG(FATAL) << "Error: unrecognized type: " << token << "!";
+      LOG(FATAL) << "[read_g2o_file] Unrecognized type: " << token << "!";
     }
 
-    // Update maximum value of poses found so far
-    size_t max_pair = std::max<double>(measurement.p1, measurement.p2);
+    // Update pose IDs
+    pose_ids.emplace(measurement.p1);
+    pose_ids.emplace(measurement.p2);
 
-    num_poses = ((max_pair > num_poses) ? max_pair : num_poses);
     measurements.push_back(measurement);
   }  // while
 
   infile.close();
 
-  num_poses++;  // Account for the use of zero-based indexing
+  // Get first pose ID
+  const size_t first_pose_id = *pose_ids.begin();
+
+  // Check for consecutive sequencing of pose IDs
+  size_t prev_pose_id = first_pose_id - 1;
```

Comment:
@adthoms This will cause overflow because `prev_pose_id` is unsigned and `first_pose_id` is nominally zero.